kms_log/main.tf

```
@@ -558,4 +593,469 @@ resource "aws_cloudwatch_metric_alarm" "dead_letter" {
+    statement {
+        actions = [
+            "sts:AssumeRole"
+        ]
+        principals {
+            type = "Service"
+            identifiers = [
+                "lambda.amazonaws.com"
+            ]
+        }
+    }
+}
+
+resource "aws_iam_role" "cloudtrail_processor" {
+    name = "${aws_lambda_function.cloudtrail_processor.id}-execution"
+    assume_role_policy = "${data.aws_iam_policy_document.assume-role.json}"
+}
+
+resource "aws_iam_role_policy" "ctprocessor_cloudwatch" {
+    name = "CloudWatch"
+    role = "${aws_iam_role.cloudtrail_processor.id}"
+    policy = "${data.aws_iam_policy_document.ctprocessor_cloudwatch.json}"
+}
+
+resource "aws_iam_role_policy" "ctprocessor_dynamodb" {
+    name = "DynamoDb"
+    role = "${aws_iam_role.cloudtrail_processor.id}"
+    policy = "${data.aws_iam_policy_document.ctprocessor_dynamodb.json}"
+}
+
+resource "aws_iam_role_policy" "ctprocessor_kms" {
+    name = "DynamoDb"
+    role = "${aws_iam_role.cloudtrail_processor.id}"
+    policy = "${data.aws_iam_policy_document.ctprocessor_kms.json}"
+}
+
+resource "aws_iam_role_policy" "ctprocessor_sns" {
+    name = "SNS"
+    role = "${aws_iam_role.cloudtrail_processor.id}"
+    policy = "${data.aws_iam_policy_document.ctprocessor_sns.json}"
+}
+
+resource "aws_iam_role_policy" "ctprocessor_sqs" {
+    name = "SQS"
+    role = "${aws_iam_role.cloudtrail_processor.id}"
+    policy = "${data.aws_iam_policy_document.ctprocessor_sqs.json}"
+}
+
+resource "aws_lambda_function" "cloudwatch_processor" {
+    count = "${var.kmslogging_service_enabled}"
+    s3_bucket = "${data.aws_s3_bucket.lambda.id}"
+    s3_key = "circleci/identity-lambda-functions/${var.lambda_identity_lambda_functions_gitrev}.zip"
+
+    lifecycle {
+        ignore_changes = ["s3_key", "last_modified"]
+    }
+
+    function_name = "${var.env_name}-kmslog-cw-processor"
+    description = "18F/identity-lambda-functions: KMS CW Log Processor"
+    role = "${aws_iam_role.cloudwatch_processor.arn}"
+    handler = "main.Functions::CloudWatchKMSHandler.process"
+    runtime = "ruby2.5"
+    timeout = 30 # seconds
+
+    environment {
+        variables = {
+            DEBUG = "${var.lambda_audit_github_debug ? "1" : ""}"
+            LOG_LEVEL = "0"
+            DDB_RETENTION_DAYS = "${var.dynamodb_retention_days}"
+            DDB_TABLE = "${aws_dynamodb_table.kms_events.id}"
+            SNS_EVENT_TOPIC_ARN = "${aws_sns_topic.kms_logging_events.arn}"
+        }
+    }
+
+    tags {
+        source_repo = "https://github.com/18F/identity-lambda-functions"
+        environment = "${var.env_name}"
+    }
+}
+
+resource "aws_lambda_event_source_mapping" "cloudwatch_processor" {
+    count = "${var.kmslogging_service_enabled}"
+    event_source_arn = "${aws_kinesis_stream.datastream.arn}"
+    function_name = "${aws_lambda_function.cloudwatch_processor.arn}"
+    starting_position = "LATEST"
+}
+
+resource "aws_iam_role" "cloudtrail_processor" {
```

```suggestion
resource "aws_iam_role" "cloudwatch_processor" {
```
aws_iam_role.cloudtrail_processor seems to be defined twice.